File: security/src/main/java/top/bettercode/summer/security/authorization/UserDetailsAuthenticationToken.java
package top.bettercode.summer.security.authorization;

import org.springframework.security.authentication.AbstractAuthenticationToken;
import org.springframework.security.core.userdetails.UserDetails;

/**
 * @author Peter Wu
 */
public class UserDetailsAuthenticationToken extends AbstractAuthenticationToken {

  private static final long serialVersionUID = 1L;
  private final UserDetails userDetails;

  public UserDetailsAuthenticationToken(UserDetails userDetails) {
    super(userDetails.getAuthorities());
    this.userDetails = userDetails;
    setAuthenticated(true);
  }

  @Override
  public Object getCredentials() {
    return "N/A";
  }

  @Override
  public Object getPrincipal() {
    return this.userDetails;
  }
}

File: security/src/main/kotlin/top/bettercode/summer/security/authorization/UserDetailsAuthenticationToken.kt
package top.bettercode.summer.security.authorization

import org.springframework.security.authentication.AbstractAuthenticationToken
import org.springframework.security.core.userdetails.UserDetails
import top.bettercode.summer.security.token.TokenId

/**
 * @author Peter Wu
 */
class UserDetailsAuthenticationToken(
    var id: TokenId,
    private val userDetails: UserDetails
) : AbstractAuthenticationToken(userDetails.authorities) {
    init {
        isAuthenticated = true
    }

    override fun getCredentials(): Any {
        return "N/A"
    }

    override fun getPrincipal(): Any {
        return userDetails
    }

}
